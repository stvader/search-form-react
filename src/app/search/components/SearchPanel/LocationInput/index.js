import LocationInput from './LocationInputContainer';

export default LocationInput;
