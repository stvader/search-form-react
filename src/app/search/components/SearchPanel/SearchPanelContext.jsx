import { createContext } from 'react';

export const SearchPanelContext = createContext(null);

export const SearchPanelContextProvider = SearchPanelContext.Provider;
