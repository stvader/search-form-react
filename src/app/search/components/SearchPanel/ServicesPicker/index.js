import ServicesPicker from './ServicesPickerContainer';

export default ServicesPicker;
