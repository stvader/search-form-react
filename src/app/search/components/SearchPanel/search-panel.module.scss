.wrapper {
  max-width: 1200px;
  width: 100%;
}

.element_wrapper {
  margin-bottom: 12px;
  width: 100%;
  padding: 0 2px;
}

.input {
  width: 100%;
}

.btn {
  margin: 7px 20px !important;
  width: calc(100% - 40px);
}

@media screen and (min-width: 750px) {
  .form_elem_wrapper {
    display: flex;
    grid-area: search;
  }

  .element_wrapper {
    margin-bottom: 0;
  }

  .btn {    
    margin-bottom: 0 !important;
    width: calc(100% - 30px);
  }
}