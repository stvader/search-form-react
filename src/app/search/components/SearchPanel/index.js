import SearchPanel from './SearchPanelContainer';

export default SearchPanel;
