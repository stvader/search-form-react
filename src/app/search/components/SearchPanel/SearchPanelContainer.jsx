import React, { memo, useMemo, useState, useCallback } from 'react';
import { compose, bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { instanceOf, object, string, func } from 'prop-types';

import { SearchPanelContextProvider } from './SearchPanelContext';
import SearchPanel from './SearchPanel';

import { selectReqQuery, selectReqDate, selectReqAddress } from '../../search.selectors';

import { setRequestData } from '../../search.actions';

import { fetchProfData } from '../../thunks/fetchData';

import { onSubmitSearchService } from '../../services/onSubmitSearchService';

import { DATE_DEFAULT } from '../../search.dateTimeConstants';

const MESSAGE_SNACKBAR_PARSE = 'Please fill all fields';

const SearchPanelContainer = ({
  history,
  location,
  query,
  savedDate,
  savedAddress,
  setRequestData,
  fetchProfData,
}) => {
  const [selectService, setSelectService] = useState(query);
  const [selectLocation, setSelectLocation] = useState(savedAddress);
  const [selectDate, setSelectDate] = useState(savedDate || DATE_DEFAULT);
  const [openSnackbar, setOpenSnackbar] = useState(false);

  const contextValue = useMemo(
    () => ({
      selectService,
      handleServiceChange: setSelectService,
      selectLocation,
      handleLocationChange: setSelectLocation,
      selectDate,
      handleDateChange: setSelectDate,
    }),
    [selectService, setSelectService, selectLocation, setSelectLocation, selectDate, setSelectDate],
  );

  const handleCloseSnackbar = useCallback((event, reason) => {
    if (reason === 'clickaway') {
      return;
    }

    setOpenSnackbar(false);
  }, []);

  const onFormSubmit = useCallback(
    e => {
      e.preventDefault();

      onSubmitSearchService({
        history,
        location,
        selectService,
        selectLocation,
        setSelectLocation,
        selectDate,
        setRequestData,
        setOpenSnackbar,
        fetchProfData,
      });
    },
    [
      history,
      location,
      selectService,
      selectLocation,
      setSelectLocation,
      selectDate,
      setRequestData,
      setOpenSnackbar,
      fetchProfData,
    ],
  );

  return (
    <SearchPanelContextProvider value={contextValue}>
      <SearchPanel
        onFormSubmit={onFormSubmit}
        openSnackbar={openSnackbar}
        handleCloseSnackbar={handleCloseSnackbar}
        messageSnackbar={MESSAGE_SNACKBAR_PARSE}
      />
    </SearchPanelContextProvider>
  );
};

const mapStateToProps = state => ({
  query: selectReqQuery(state),
  savedDate: selectReqDate(state),
  savedAddress: selectReqAddress(state),
});

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      setRequestData,
      fetchProfData,
    },
    dispatch,
  );

SearchPanelContainer.propTypes = {
  history: object,
  location: object,
  query: string,
  savedDate: instanceOf(Date),
  savedAddress: object,
  setRequestData: func,
  fetchProfData: func,
};

export default compose(
  withRouter,
  connect(mapStateToProps, mapDispatchToProps),
)(memo(SearchPanelContainer));
