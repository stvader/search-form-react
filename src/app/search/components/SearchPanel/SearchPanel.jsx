import React from 'react';
import { func, string } from 'prop-types';
import { Button, Snackbar } from '@material-ui/core';
import MuiAlert from '@material-ui/lab/Alert';

import ServicesPicker from './ServicesPicker';
import LocationInput from './LocationInput';
import DatePicker from './DatePicker';

import s from './search-panel.module.scss';

const Alert = props => <MuiAlert elevation={6} variant="filled" {...props} />;

const SearchPanel = ({ onFormSubmit, openSnackbar, handleCloseSnackbar, messageSnackbar }) => (
  <div className={s.wrapper}>
    <form onSubmit={onFormSubmit}>
      <div className={s.form_elem_wrapper}>
        <div className={s.element_wrapper}>
          <ServicesPicker />
        </div>
        <div className={s.element_wrapper}>
          <LocationInput />
        </div>
        <div className={s.element_wrapper}>
          <DatePicker />
        </div>

        <div className={s.element_wrapper}>
          <Button type="submit" className={s.btn} variant="contained" color="primary" size="large">
            Search
          </Button>
        </div>
      </div>
    </form>
    <Snackbar
      anchorOrigin={{
        vertical: 'bottom',
        horizontal: 'left',
      }}
      open={openSnackbar}
      autoHideDuration={6000}
      onClose={handleCloseSnackbar}
    >
      <Alert onClose={handleCloseSnackbar} severity="error">
        {messageSnackbar}
      </Alert>
    </Snackbar>
  </div>
);

SearchPanel.propTypes = {
  onFormSubmit: func,
  openSnackbar: func,
  handleCloseSnackbar: func,
  messageSnackbar: string,
};

export default SearchPanel;
