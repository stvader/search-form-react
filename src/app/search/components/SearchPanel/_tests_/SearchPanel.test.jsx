import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Button } from '@material-ui/core';

import SearchPanel from '../SearchPanel';

configure({ adapter: new Adapter() });

describe('SearchPanel', () => {
  it('SearchPanel render', () => {
    const component = shallow(<SearchPanel />);

    const form = component.find('form');
    const wrapper = component.find('.search-panel-form-elem-wrapper');
    const btn = component.find(Button);

    expect(component).toMatchSnapshot();
    expect(form).toBeDefined();
    expect(wrapper.children()).toHaveLength(4);
    expect(btn).toBeDefined();
    expect(btn.prop('type')).toBe('submit');
  });
});
