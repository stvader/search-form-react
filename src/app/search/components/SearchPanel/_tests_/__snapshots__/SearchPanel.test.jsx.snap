// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPanel SearchPanel render 1`] = `ShallowWrapper {}`;
